test/0.8.9/staking-router/report-exited-keys.test.js

```diff
@@ -0,0 +1,456 @@
+    }
+  })
+
+  describe('Report exited keys by module changes stake allocation', async () => {
+    before(async () => {
+      // add modules
+      await router.addStakingModule(
+        'Module 1',
+        operators.address,
+        10_000, // 100 % _targetShare
+        500, // 10 % _moduleFee
+        500, // 50 % _treasuryFee
+        { from: admin }
+      )
+      module1Id = +(await router.getStakingModuleIds())[0]
+
+      await router.addStakingModule(
+        'Module 2',
+        module2.address,
+        10_000, // 100 % _targetShare
+        500, // 10 % _moduleFee
+        500, // 50 % _treasuryFee
+        { from: admin }
+      )
+      module2Id = +(await router.getStakingModuleIds())[1]
+
+      // add operators
+      await operators.testing_addNodeOperator(
+        'Operator1',
+        ADDRESS_1, // config.rewardAddress,
+        100, // totalSigningKeysCount,
+        60, // vettedSigningKeysCount,
+        50, // depositedSigningKeysCount,
+        0 // exitedSigningKeysCount
+      )
+      await operators.testing_addNodeOperator(
+        'Operator1',
+        ADDRESS_1, // config.rewardAddress,
+        100, // totalSigningKeysCount,
+        60, // vettedSigningKeysCount,
+        50, // depositedSigningKeysCount,
+        0 // exitedSigningKeysCount
+      )
+
+      await module2.testing_addNodeOperator(
+        'Operator1',
+        ADDRESS_1, // config.rewardAddress,
+        20, // totalSigningKeysCount,
+        15, // vettedSigningKeysCount,
+        10, // depositedSigningKeysCount,
+        0 // exitedSigningKeysCount
+      )
+    })
+
+    beforeEach(snapshot)
+    afterEach(revert)
+
+    it('check initial keys of operators', async () => {
+      const moduleSummary1 = await router.getNodeOperatorSummary(module1Id, 0)
+      assert.equal(moduleSummary1.isTargetLimitActive, false)
+      assert.equal(moduleSummary1.targetValidatorsCount, 0)
+      assert.equal(moduleSummary1.stuckValidatorsCount, 0)
+      assert.equal(moduleSummary1.refundedValidatorsCount, 0)
+      assert.equal(moduleSummary1.stuckPenaltyEndTimestamp, 0)
+      assert.equal(moduleSummary1.totalExitedValidators, 0)
+      assert.equal(moduleSummary1.totalDepositedValidators, 50)
+      assert.equal(moduleSummary1.depositableValidatorsCount, 10)
+
+      const moduleSummary2 = await router.getNodeOperatorSummary(module2Id, 0)
+      assert.equal(moduleSummary2.isTargetLimitActive, false)
+      assert.equal(moduleSummary2.targetValidatorsCount, 0)
+      assert.equal(moduleSummary2.stuckValidatorsCount, 0)
+      assert.equal(moduleSummary2.refundedValidatorsCount, 0)
+      assert.equal(moduleSummary2.stuckPenaltyEndTimestamp, 0)
+      assert.equal(moduleSummary2.totalExitedValidators, 0)
+      assert.equal(moduleSummary2.totalDepositedValidators, 10)
+      assert.equal(moduleSummary2.depositableValidatorsCount, 5)
+    })
+
+    it('report exited keys should change rewards distribution', async () => {
```

Comment:
Do we have a test for the following scenario?

### oracle round i

oracle reports total exited keys via `updateExitedValidatorsCountByStakingModule` (let's say, `M`)
oracle reports per-operator exited keys only partially (due to limits) via `reportStakingModuleExitedValidatorsCountByNodeOperator` (let's say, `N < M`)

### oracle round i+1

oracle reports new total exited keys via `updateExitedValidatorsCountByStakingModule` (let's say, `M2`)
oracle reports per-operator exited keys both for the current and previous round (limits allow) via `reportStakingModuleExitedValidatorsCountByNodeOperator` (let's say, `N2 == (N-M) + M2) > M2`)

It was raised by the oracle daemon development subteam.